### PR TITLE
Add profileCustomizations field to KubeDescheduler resource

### PR DESCRIPTION
**Pull request by RoniKishner**
##### Short description:
Add profileCustomizations field to KubeDescheduler resource

##### More details:
This field is needed for special features for example [enabling devEnableEvictionsInBackground](https://github.com/openshift/cluster-kube-descheduler-operator/pull/370) which is still in alpha stage

https://github.com/ingvagabund/cluster-kube-descheduler-operator/blob/3d95d0b3ed7962b91ef74e8e37661a09fba47a2f/README.md#profile-customizations

##### Which issue(s) this PR fixes:
https://issues.redhat.com/browse/CNV-47165

## Summary by CodeRabbit

- **New Features**
	- Introduced several new optional parameters in the KubeDescheduler class, including `descheduling_interval_seconds`, `observed_config`, and `unsupported_config_overrides`, enhancing configurability.
	- Added new JSON schemas for `CloudPrivateIPConfig` and `KubeDescheduler`, improving resource management capabilities.
	- Enhanced the console UI with a new "capabilities" property for better user interaction.
- **Bug Fixes**
	- Improved validation rules across various schemas to ensure data integrity and compliance.
- **Documentation**
	- Expanded documentation for new parameters and properties in the KubeDescheduler class and associated schemas for better clarity.

### CODE REVIEW

**Comment by myakove** [comment]
@RoniKishner  
can you run `class-generator` for this kind and see if we can use the generated one over the existing one?

**Comment by RoniKishner** [comment]
> class-generator

After running --update-schema it still wouldn't create the scheme and gave an error:
`ubedescheduler not found in class_generator/schema/__resources-mappings.json`

**Comment by myakove** [comment]
> > class-generator
> 
> After running --update-schema it still wouldn't create the scheme and gave an error: `ubedescheduler not found in class_generator/schema/__resources-mappings.json`

did you run it from inside this repo directory?
Do you get output when you run `oc api-resources  | grep -w KubeDescheduler` ?

**Comment by RoniKishner** [comment]
I forget I had to install the deschduler operator for the CRD, that's why it didn't work.
In any case it looks like the class_generator has the exact same code.
I'm not sure about adding the resource to the schema files though because it is a resource which needs an installation.

**Comment by myakove** [comment]
@rkishner Please update the schema and use the generated file.

File: class_generator/tests/test_class_generator.py [hunk 1]
```diff
@@ -16,12 +16,13 @@
         "ConfigMap",
         "DNS",
         "Deployment",
+        "Image",
```
Comment (myakove) [inline class_generator/tests/test_class_generator.py:19]: No need to add a test for this resource.

**Review by myakove — CHANGES_REQUESTED** [review]

File: class_generator/tests/test_class_generator.py (hunk 1 above)
Comment (RoniKishner) [reply class_generator/tests/test_class_generator.py:19]: Done

**Review by myakove — APPROVED** [review]

**Pull request merged**